api/arceos_posix_api/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+#![cfg_attr(all(not(test), not(doc)), no_std)]
+#![feature(ip_in_core)]
+#![feature(int_roundings)]
+#![feature(naked_functions)]
+#![feature(result_option_inspect)]
+#![allow(clippy::missing_safety_doc)]
+#![allow(missing_docs)]
+#![allow(non_camel_case_types, non_upper_case_globals)]
+
+extern crate axruntime;
+
+#[macro_use]
+extern crate axlog;
+
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
+#[macro_use]
+mod utils;
+
+mod imp;
+
+/// cbindgen:ignore
+#[rustfmt::skip]
+#[path = "./ctypes_gen.rs"]
+#[allow(dead_code, non_snake_case, non_camel_case_types, non_upper_case_globals, clippy::upper_case_acronyms)]
+pub mod ctypes;
+
+pub use imp::sync::*;
```

Comment:
These items should not be exported.

Reply:
`imp::sync::Mutex` is used in `println_str` to support atomically print messages and `'\n'`(`println_str` is used in `puts`). 
This is actually not a proper implementation, but `println_str` needs a mutex. How to solve this?

Reply:
1. Implement `puts` in rust.
2. Call ` pthread_mutex_*` (C style locks) in ` puts`.

Future: implement ` sys_futex` in `arceos_posix_api`, and use futex to implement rust style locks in ` axlibc`.